test_conformance/subgroups/test_workgroup_extended_types.cpp

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "procs.h"
+#include "subhelpers.h"
+#include "workgroup_common_kernels.h"
+#include "workgroup_common_templates.h"
+#include "harness/typeWrappers.h"
+
+struct run_for_type
+{
+    run_for_type(cl_device_id device, cl_context context,
+                 cl_command_queue queue, int num_elements,
+                 bool useCoreSubgroups,
+                 std::vector<std::string> required_extensions = {})
```

Comment:
Looking at where this object will ultimately be constructed, it seems this should be a vector of const char * to avoid the std::string construction.
Alternative, use std::string_view but that requires upping the C++ standard requirements to 17.